Algolia.Search/Utils/JsonHelper.cs

```diff
@@ -0,0 +1,79 @@
+/*
+* Copyright (c) 2018 Algolia
+* http://www.algolia.com/
+* Based on the first version developed by Christopher Maneu under the same license:
+*  https://github.com/cmaneu/algoliasearch-client-csharp
+* 
+* Permission is hereby granted, free of charge, to any person obtaining a copy
+* of this software and associated documentation files (the "Software"), to deal
+* in the Software without restriction, including without limitation the rights
+* to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+* copies of the Software, and to permit persons to whom the Software is
+* furnished to do so, subject to the following conditions:
+* 
+* The above copyright notice and this permission notice shall be included in
+* all copies or substantial portions of the Software.
+* 
+* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+* IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+* FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+* AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+* LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+* OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+* THE SOFTWARE.
+*/
+
+using System.IO;
+using System.Text;
+using System.Threading.Tasks;
+using Newtonsoft.Json;
+
+namespace Algolia.Search.Utils
+{
+    /// <summary>
+    /// Helper to improve performance with large json
+    /// https://www.newtonsoft.com/json/help/html/performance.htm
+    /// </summary>
+    public static class JsonHelper
+    {
+        public static void Serialize<T>(T data, Stream stream, JsonSerializerSettings settings)
+        {
+            using (var sw = new StreamWriter(stream, new UTF8Encoding(false), 1024, true))
```

Comment:
Better to move UTF8Encoding initialisation to a static field outside of the method rather than newing up per serialise call. Recycling 👍

Reply:
@ChrisMcKee By the way, have you tried/looked this alpha version yet? I would appreciate having users point of view on it.
Thanks!

Reply:
Nope; its on my list. I was planning on gutting a POC from when we first started using algolia and shoving this client in to see how it performs in comparison (Write load). We have a few readonly instances I want to compare memory usage against as well. 

We have cases where our actions are against N indexed (not slave indexes) but in order to do that we have to create N clients at the moment (even though the only real action change is on the InitClient in order to switch which index a query hits).  

The InitClient requirement is a PITA